packages/pkp-ethers/src/lib/helper.ts

```diff
@@ -24,8 +24,10 @@ export function convertHexToUtf8(value: string): string {
  * @param {any} txParams - The original transaction parameters.
  * @returns {any} The transaction object with 'gas' and 'from' fields removed.
  */
-export const getTransactionToSign = (txParams: any): any => {
-  let formattedTx = Object.assign({}, txParams);
+export const getTransactionToSign = <T>(
+  txParams: T & { gas?: string; from?: string }
```

Comment:
I think we can use the `UnsignedTransaction` type?

Reply:
Kind of, the source of the param that reaches this function is a `TransactionRequest` or an `any` nested inside a `ETHRequestSigningPayload`. So, it feels like yes, we should use that type, but in reality it does not comply with the function code. That is because `UnsignedTransaction` has `gasLimit` and `gasPrice`, not `gas`

Considering the function deletes `gas` when it is there, I would say the function has to be checked to have the right logic we want (deleting `gas` instead of `gasSomething`) and I think we better do those kind of things in other tickets, if needed, to not add extra delay and complexity